### PR TITLE
Improve dev setup instructions

### PR DESCRIPTION
**Pull request by in0rdr**
Setting up a new dev environment on my local machine. This PR documents the changes I would like to see regarding the setup from scratch.

- Dev files are not generated if dev mode is not activated
- Some more explanation is needed regarding the certificate to validate the JWT keys. The keys need to be placed on Distributey and the Vault server side.

### CODE REVIEW

**Comment by p15r** [comment]
@in0rdr my GH notifications didn't work - sorry for the delayed response. What actually changed in docs/cache-only-key-service-v3.xml?

File: docs/key_consumer_setup.md [hunk 1]
```diff
@@ -24,6 +24,15 @@ Configure Salesforce to authenticate its HYOK requests to `distributey` using JW
        - `Unique Name`: this is the `KID` of the `JWT` token, thus must be unique. Recommended naming scheme: `jwt_kid_salesforce_serviceX`
      - Download the certificate and save it by its `Unique Name`. It must later be configured in `distributey` (`config/auth/`).
      - Salesforce provides a certificate from which `distributey` only needs its public key. Use this command to extract it: `openssl x509 -pubkey -noout -in jwt_kid_salesforce_serviceX.crt > jwt_kid_salesforce_serviceX.pub`
+     - Store the same public `jwt_kid_salesforce_serviceX.pub` key on the Vault JWT auth backend as part of `auth_jwt_validation_pubkeys` in the Terraform variables ([`./docker/terraform/terraform.tfvars`](../docker/terraform/terraform.tfvars)). For a production setup, this configuration needs to be done on the Vault server side, not on the development Vault in the Docker container:
```
Comment (p15r) [inline docs/key_consumer_setup.md:27]: Can you mention that this is done (automatically) by `dev/write_jwt_to_tfvars.py` which is called by `dev/dev_setup.sh`?

I appreciate the callout about the production setup!

**Review by p15r — CHANGES_REQUESTED** [review]
Great input 🙏

**Comment by in0rdr** [comment]
> @in0rdr my GH notifications didn't work - sorry for the delayed response. What actually changed in docs/cache-only-key-service-v3.xml?

I removed a reference to AppRole authentication from the picture. It's JWT only authenticaiton, all the way (from src to dest).

**Pull request merged**